astropy/io/fits/src/compressionmodule.c

```diff
@@ -764,6 +768,13 @@ void init_output_buffer(PyObject* hdu, void** buf, size_t* bufsize) {
     }
 
     *buf = calloc(*bufsize, sizeof(char));
+    if (*buf == NULL) {
+      // Checking if calloc failed.
+      PyErr_SetString(PyExc_TypeError,
```

Comment:
This should be a `PyExc_MemoryError` and the next line has inconsistent indentation.